src/Common/src/System/Net/Logging/HttpEventSource.cs

```diff
@@ -0,0 +1,161 @@
+        {
+            const int SIZEDATA = 4;
+            fixed (char* arg1Ptr = objectA, arg2Ptr = objectB)
+            {
+                EventData* dataDesc = stackalloc EventSource.EventData[SIZEDATA];
+
+                dataDesc[0].DataPointer = (IntPtr)(arg1Ptr);
+                dataDesc[0].Size = (objectA.Length + 1) * sizeof(char); // Size in bytes, including a null terminator. 
+                dataDesc[1].DataPointer = (IntPtr)(&objectAHash);
+                dataDesc[1].Size = sizeof(int);
+                dataDesc[2].DataPointer = (IntPtr)(arg2Ptr);
+                dataDesc[2].Size = (objectB.Length + 1) * sizeof(char);
+                dataDesc[3].DataPointer = (IntPtr)(&objectBHash);
+                dataDesc[3].Size = sizeof(int);
+
+                WriteEventCore(ASSOCIATE_ID, SIZEDATA, dataDesc);
+            }
+        }
+
+        [NonEvent]
+        internal static void UriBaseAddress(object obj, string baseAddress)
+        {
+            if (!s_log.IsEnabled())
+            {
+                return;
+            }
+            s_log.UriBaseAddress(baseAddress, LoggingHash.GetObjectName(obj), LoggingHash.HashInt(obj));
+        }
+
+        [Event(URI_BASE_ADDRESS_ID, Keywords = Keywords.Debug,
+            Level = EventLevel.Informational)]
+        internal unsafe void UriBaseAddress(string uriBaseAddress, string objName, int objHash)
+        {
+            const int SIZEDATA = 3;
+            fixed (char* arg1Ptr = uriBaseAddress, arg2Ptr = objName)
+            {
+                EventData* dataDesc = stackalloc EventSource.EventData[SIZEDATA];
+
+                dataDesc[0].DataPointer = (IntPtr)(arg1Ptr);
+                dataDesc[0].Size = (uriBaseAddress.Length + 1) * sizeof(char); // Size in bytes, including a null terminator. 
+                dataDesc[1].DataPointer = (IntPtr)(arg2Ptr);
+                dataDesc[1].Size = (objName.Length + 1) * sizeof(char);
+                dataDesc[2].DataPointer = (IntPtr)(&objHash);
+                dataDesc[2].Size = sizeof(int);
+
+                WriteEventCore(URI_BASE_ADDRESS_ID, SIZEDATA, dataDesc);
+            }
+        }
+
+        [NonEvent]
+        internal static void ContentNull(object obj)
+        {
+            if (!s_log.IsEnabled())
+            {
+                return;
+            }
+            s_log.ContentNull(LoggingHash.GetObjectName(obj), LoggingHash.HashInt(obj));
+        }
+
+        [Event(CONTENT_NULL_ID, Keywords = Keywords.Debug,
+            Level = EventLevel.Informational)]
+        internal void ContentNull(string objName, int objHash)
+        {
+            WriteEvent(CONTENT_NULL_ID, objName, objHash);
+        }
+
+        [NonEvent]
+        internal static void ClientSendCompleted(HttpClient httpClient, HttpResponseMessage response, HttpRequestMessage request)
+        {
+            if (!s_log.IsEnabled())
+            {
+                return;
+            }
+            string responseString = "";
```

Comment:
Nit: New line after closing } if followed by a statement.
Throughout the files (will stop adding comments for this issue).